samples/cpp/stereo_calib.cpp

```diff
@@ -99,8 +114,29 @@ StereoCalib(const vector<string>& imagelist, Size boardSize, float squareSize, b
                     timg = img;
                 else
                     resize(img, timg, Size(), scale, scale, INTER_LINEAR_EXACT);
-                found = findChessboardCorners(timg, boardSize, corners,
-                    CALIB_CB_ADAPTIVE_THRESH | CALIB_CB_NORMALIZE_IMAGE);
+
+                if (type == "chessboard") {
+                    found = findChessboardCorners(timg, boardSize, corners,
+                        CALIB_CB_ADAPTIVE_THRESH | CALIB_CB_NORMALIZE_IMAGE);
+                }
+                else {
+                    cv::aruco::CharucoBoard ch_board({ boardSize.width + 1, boardSize.height + 1 },
+                        squareSize, markerSize, dictionary);
+
+                    std::vector<int> markerIds;
+
+                    cv::aruco::CharucoDetector ch_detector(ch_board);
+                    ch_detector.detectBoard(timg, corners, markerIds);
```

Comment:
Also the following cornerSubPix does not make sense for Charco markers. They are already refined.